### ISSUE TITLE
Support latest analyzer version from Flutter 3.19.0

### ISSUE THREAD

**Issue by marcinkaczmarzyk** [isar/isar#1564]
### Steps to Reproduce

Run build runner and there will be warning about that it cannot use same version of analyzer as the flutter uses it

Because no versions of isar_generator match >3.1.0+1 <4.0.0 and isar_generator 3.1.0+1 depends on analyzer >=4.6.0 <6.0.0, isar_generator ^3.1.0+1 requires analyzer >=4.6.0 <6.0.0.

### Details

Flutter 3.19.0
isar_generator: ^3.1.0+1

**Comment by Bohne13** [comment]
Did you try updating your dependencies?
```shell
flutter pub upgrade
``` 
and if this doesn't help:
```shell
flutter pub outdated
``` 

what also could help is removing the dependency of 'analyzer' from the 'pubspec.yaml'. If you run
```shell
flutter pub get
``` 
flutter resolves the dependencies by itself.

I hope that helps!

**Comment by huahua8893** [comment]
add this to your pubspec.yaml:
dependency_overrides:
  analyzer: ^6.3.0

**Comment by mrclauss** [comment]
This is fixed in isar-community fork

**Comment by gmcdowell** [comment]
Likely a silly question: Can you clarify how to configure in pubspec.yaml to refer to community fork  of `isar_generator` please?

I have tried the following:

```
dev_dependencies:
...
isar_generator: 
    git: 
      url: https://github.com/isar-community/isar.git
      ref: 3.1.0+1
```

**Comment by mrclauss** [comment]
Yes, or you may specify the new tag that was just created. The readme on the fork now shows how to do this.

**Comment by petersnoopy** [comment]
Any updates for the official repo?

**Comment by vicenterusso** [comment]
> Any updates for the official repo?

No, Simon is silent. That why we are keeping the [community fork](https://github.com/isar-community/isar) alive

**Comment by zraki-kenpachi** [comment]
this is the same as #1625

**Comment by gonzalogauto** [comment]
Same issue when updating build_runner and build packages to the latest version.

`
Because isar_generator 3.1.0+1 depends on analyzer >=4.6.0 <6.0.0 and build >=2.4.2 depends on analyzer >=6.9.0 <8.0.0, isar_generator 3.1.0+1 is incompatible with build >=2.4.2.
And because build_runner >=2.5.2 depends on build 2.5.2, isar_generator 3.1.0+1 is incompatible with build_runner >=2.5.2.
`

**Comment by gonzalogauto** [comment]
Any updates? @simc
